scripts/checkpoint_converters/convert_llava_hf_to_nemo.py

```diff
@@ -0,0 +1,328 @@
+    model_config["num_query_groups"] = ref_config["num_key_value_heads"]
+    model_config["layernorm_epsilon"] = ref_config["rms_norm_eps"]
+    model_config["init_method_std"] = ref_config["initializer_range"]
+    model_config["kv_channels"] = ref_config.get(
+        "head_dim", model_config["hidden_size"] // model_config["num_attention_heads"]
+    )
+    if ref_config.get("rope_scaling") is not None:
+        if ref_config["rope_scaling"]["type"] == "linear":
+            model_config["seq_len_interpolation_factor"] = ref_config["rope_scaling"]["factor"]
+        else:
+            raise ValueError("Only linear rope scaling type is supported now")
+    model_config["use_cpu_initialization"] = True
+
+    return model_config
+
+
+def get_args():
+    parser = ArgumentParser()
+    parser.add_argument("--input_name_or_path", type=str)
+    parser.add_argument("--tokenizer_path", type=str)
+    parser.add_argument("--conv_template", default="v1", type=str)
+    parser.add_argument(
+        "--hparams_file",
+        type=str,
+        default=os.path.join(
+            os.path.dirname(__file__), '../../examples/multimodal/multimodal_llm/neva/conf/llava_config.yaml'
+        ),
+        required=False,
+        help="Path config for restoring. It's created during training and may need to be modified during restore if restore environment is different than training. Ex: /raid/nemo_experiments/megatron_gpt/hparams.yaml",
+    )
+    parser.add_argument("--output_path", type=str, default=None, help="Path to output .nemo file.")
+    parser.add_argument(
+        "--precision", type=str, default="bf16", choices=["bf16", "32"], help="Precision for checkpoint weight saved"
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def convert(args):
+    logging.info(f"Loading checkpoint from HF Llava: `{args.input_name_or_path}`")
+    hf_tokenizer = LlamaTokenizer.from_pretrained(args.input_name_or_path)
+    hf_model = LlavaForConditionalGeneration.from_pretrained(args.input_name_or_path)
+    logging.info("HF Model loading done.")
+
+    nemo_config = OmegaConf.load(args.hparams_file)
+    nemo_config.model = adjust_nemo_config(nemo_config.model, hf_model.config.__dict__)
+    nemo_config.model.data["conv_template"] = args.conv_template
+    nemo_config.model.mm_cfg.llm["model_type"] = args.conv_template
+    nemo_config.model.tokenizer["model"] = args.tokenizer_path
+
+    nemo_config.trainer["precision"] = args.precision
+    trainer = MegatronTrainerBuilder(nemo_config).create_trainer()
+    model = MegatronNevaModel(nemo_config.model, trainer)
+
+    rename_keys = create_rename_keys(nemo_config.model.num_layers)
+    old_state_dict = hf_model.state_dict()
+    new_state_dict = rename_model_keys(model_state_dict=old_state_dict, rename_keys=rename_keys)
+
+    nemo_state_dict = adjust_tensor_shapes(model, new_state_dict)
+    model.load_state_dict(nemo_state_dict, strict=False)
+
+    logging.info(f'=' * 100)
+    # Verifications
```

Comment:
can we make this verification optional for users on CPU or smaller GPUs?
same thing in gemma conversion script

Reply:
I will just add a flag?